ext4: prevent a fs without journal from being resized

This patch prevents a fs without journal from being resized, because
it is easy to detroy the fs.

Signed-off-by: Yongqiang Yang <xiaoqiangnk@gmail.com>
diff --git a/fs/ext4/resize.c b/fs/ext4/resize.c
index 53d9795..33ab40d 100644
--- a/fs/ext4/resize.c
+++ b/fs/ext4/resize.c
@@ -33,6 +33,16 @@ int ext4_resize_begin(struct super_block *sb)
 		return -EPERM;
 	}
 
+	/*
+	 * We are not allowed to do online-resizing on a filesystem without
+	 * journal, otherwise, it is easy to destroy the filesystem.
+	 */
+	if (!EXT4_SB(sb)->s_journal) {
+		ext4_warning(sb, "There is no journal for the filesystem, "
+			     "so online resizing is not allowed\n");
+		return -EPERM;
+	}
+
 	if (test_and_set_bit_lock(EXT4_RESIZING, &EXT4_SB(sb)->s_resize_flags))
 		ret = -EBUSY;
 
